### PR TITLE
[bitnami/haproxy] Adding condition to serviceaccount

### PR DESCRIPTION
**Pull request by GGre4**
**Description of the change**

Adding serviceAccount.create = false in the values was being ignored and creating a new service account each time a new deployment was made.  

- Added condition to the service account template

### CODE REVIEW

File: bitnami/haproxy/templates/service-account.yaml [hunk 1]
```diff
@@ -1,13 +1,21 @@
+{{- if .Values.serviceAccount.create }}
```
Comment (pablogalegoc) [inline bitnami/haproxy/templates/service-account.yaml:1]: I would keep just this condition and remove the unrelated changes

File: bitnami/haproxy/templates/service-account.yaml [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+  {{- if or .Values.commonAnnotations .Values.serviceAccount.annotations }}
+  annotations:
+    {{- if or .Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.serviceAccount.annotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.serviceAccount.annotations "context" $ ) | nindent 4 }}
+    {{- end }}
   {{- end }}
+{{- end }}
```
Comment (pablogalegoc) [inline bitnami/haproxy/templates/service-account.yaml:21]: New line need here

**Review by pablogalegoc — CHANGES_REQUESTED** [review]
Hi @GGre4! Thanks for your contribution! I left some comments below. Also remember that a patch version bump of the chart will be needed to pass the lint test 🙂

File: bitnami/haproxy/templates/service-account.yaml (hunk 1 above)
Comment (GGre4) [reply bitnami/haproxy/templates/service-account.yaml:1]: Hi @pablogalegoc Thanks for your feedback, first time contributing to a project. 

I have updated the PR accordingly

Comment (pablogalegoc) [reply bitnami/haproxy/templates/service-account.yaml:1]: Hey @GGre4, that's awesome! This kind of little changes are very useful and help building confidence.
I asked to keep the changes just to what's intended for the PR because then it's easier to find them and reason about them 🙂

File: bitnami/haproxy/templates/service-account.yaml (hunk 2 above)
Comment (pablogalegoc) [reply bitnami/haproxy/templates/service-account.yaml:15]: @GGre4 remember to add a new line here. If you are using the Github editor it sometimes messes with the endlines, it is better to change it locally and push

Comment (pablogalegoc) [reply bitnami/haproxy/templates/service-account.yaml]: Yep, that's it

**Review by pablogalegoc — APPROVED** [review]
Thank you @GGre4! LGTM!
